cmd/clusterctl/test/e2e/helpers.go

```diff
@@ -0,0 +1,150 @@
+// +build e2e
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	appsv1 "k8s.io/api/apps/v1"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	clusterctlclient "sigs.k8s.io/cluster-api/cmd/clusterctl/pkg/client"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/test/framework/management/kind"
+	infrav1 "sigs.k8s.io/cluster-api/test/infrastructure/docker/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func CreateKindCluster(configFile string) (*kind.Cluster, error) {
+	var kindCluster *kind.Cluster
+	var err error
+	framework.TryAddDefaultSchemes(clientgoscheme.Scheme)
+	Expect(infrav1.AddToScheme(clientgoscheme.Scheme)).To(Succeed())
+
+	// create kind cluster and client
+	kindClusterName := "clusterctl-e2e-test-" + util.RandomString(6)
+	if configFile == "" {
+		kindCluster, err = kind.NewCluster(ctx, kindClusterName, clientgoscheme.Scheme, managerImage)
```

Comment:
`ctx` and `managerImage` appear to be undefined...am i missing a global somewhere?

Reply:
They are in `BeforeSuite(func() `

Reply:
ah yeah, weird, my editor is just being dumb, thanks